addons/account/models/account_journal.py

```diff
@@ -323,6 +323,21 @@ def _check_type_default_account_id_type(self):
             if journal.type in ('sale', 'purchase') and journal.default_account_id.user_type_id.type in ('receivable', 'payable'):
                 raise ValidationError(_("The type of the journal's default credit/debit account shouldn't be 'receivable' or 'payable'."))
 
+    @api.constrains('active')
+    def _check_auto_post_draft_entries(self):
+        for journal in self:
+            pending_moves = self.env['account.move'].search([
+                ('journal_id', '=', journal.id),
+                ('state', '=', 'draft')
+            ])
```

Comment:
add `limit=1` to not force the orm to fetch everything for nothing ;)

Reply:
Yep indeed sorry about that

Reply:
or just `search_count`